public/kibi_timeline.js

```diff
@@ -191,112 +191,119 @@ define(function (require) {
             const uniqueLabels = [];
 
             _.each(searchResp.hits.hits, function (hit) {
-              let labelValue = timelineHelper.pluckLabel(hit, params, notify);
-              if (labelValue.constructor === Array) {
-                labelValue = labelValue.join(', ');
+              let labelValues = timelineHelper.pluckLabel(hit, params, notify);
+
+              if (!params.splitOnArray && labelValues.constructor === Array) {
+                labelValues = labelValues.join(', ');
               }
 
-              if (params.startFieldSequence) { // in kibi, we have the path property of a field
-                startFieldValue = kibiUtils.getValuesAtPath(hit._source, params.startFieldSequence);
-              } else {
-                startFieldValue = _.get(hit._source, params.startField);
-                if (startFieldValue && startFieldValue.constructor !== Array) {
-                  startFieldValue =  [ startFieldValue ];
-                }
+              if (labelValues.constructor !== Array) {
```

Comment:
As the constructor changes in the condition preceding the condition that creates an array, that is not preferable.